### PR TITLE
Added build section

### PR DESCRIPTION
**Pull request by JesperIRL**
Instructions on how to build the JDK.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - Author)
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/guide pull/56/head:pull/56` \
`$ git checkout pull/56`

Update a local copy of the PR: \
`$ git checkout pull/56` \
`$ git pull https://git.openjdk.java.net/guide pull/56/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 56`

View PR using the GUI difftool: \
`$ git pr show -t 56`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/guide/pull/56.diff">https://git.openjdk.java.net/guide/pull/56.diff</a>

</details>

### CODE REVIEW

File: src/index.md [hunk 1]
```diff
@@ -981,6 +981,77 @@ For the purposes of brevity this document will use the term "bug" to refer to bo
...
+:::
+
+The JDK build system is a fairly complex machine that has the ability to build anything from a single module to a complete shippable JDK bundle with various levels of debug capabilities, run tests, install your newly built JDK on your system, or cross-compile for some other system. The build uses `make` and a few other tools that you will have to install on your system before starting.
+
+The JDK supports incremental builds. This means that if you have a complete build and make changes in just a single part of the JDK (e.g. a module or part of the JVM), only that particular part needs to be rebuilt. So subsequent builds will be faster and you can always use a make target that results in a complete JDK image without having to worry about actually building the entire JDK every time. Please note that the incremental build do have limmits in its understanding of what you change. For instance, if you change behaviors or conventions in one module there may be other parts of the JDK that implicitly depends on these without make's knowledge. For this reason you may have to rebuild several modules, or do a clean build if you change things that may have a wider impact.
```
Comment (dholmes-ora) [inline src/index.md:995]: typo: limmits

File: src/index.md [hunk 2]
```diff
@@ -981,6 +981,77 @@ For the purposes of brevity this document will use the term "bug" to refer to bo
...
+
+The log below shows the steps taken to download and build the JDK source code from the mainline JDK development GIT repository. The configure script will tell you what additional packages you need. In this particular case several packages was needed since this build was performed on a clean Ubuntu installation. The configure script was run several times to get all the dependencies, but only the commands actually needed to get the JDK built is included in the log.
```
Comment (dholmes-ora) [inline src/index.md:997]: s/was needed/were needed/
s/is included/are included/

File: src/index.md [hunk 3]
```diff
@@ -981,6 +981,77 @@ For the purposes of brevity this document will use the term "bug" to refer to bo
...
+| Target | What it does      |
+|:-------|:----------|
+| `exploded-image` | This is the default make target that you'll get if you simply invoke `make`. |
+| `image` | Builds a complete JDK image. A good target to use if you want to build a JDK for general usage or if you want to test something closer to the shipping product. This can also be a good target to use if doing something which might have a build aspect to it. |
+| `<name>-image` | Build just the image for any of jdk, test, docs, symbols, etc. |
+| `reconfigure` | Re-runs the configure script with the same arguments as given the last time. |
+| `demos` | Builds the demos which for instance make it easy to test something UI related. |
+| `docs` | Builds the javadoc. Note that a number of classes in the javadoc API are generated during the build, so `make docs` might do more than simply invoke `javadoc`, depending on the state of your build. |
+| `java.base` | Builds the base module. You can (re)build any module with `make <module>`. |
+| `hotspot` | Builds only the JVM. |
```
Comment (dholmes-ora) [inline src/index.md:1049]: And any dependencies, which includes quite a bit actually.

**Review by dholmes-ora — APPROVED** [review]
Seems okay though I'm not sure how much can/should go in this guide. I don't see any link to the actual build docs though.

Thanks,
David

File: src/index.md (hunk 1 above)
Comment (JesperIRL) [reply src/index.md]: Fixed.

File: src/index.md (hunk 2 above)
Comment (JesperIRL) [reply src/index.md]: Fixed.

File: src/index.md (hunk 3 above)
Comment (JesperIRL) [reply src/index.md]: Rephrased.

**Comment by JesperIRL** [comment]
Thank you David! New version should address all your comments.

**Review by dholmes-ora — APPROVED** [review]
LGTM! Thanks Jesper.

David

**Review by irisclark — APPROVED** [review]

**Pull request closed without merging**